portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,44 +33,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-// List of comments the portfolio received.
- private List<Comment> comments;
 
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-  }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json");
-    String json = new Gson().toJson(comments);
-    response.getWriter().println(json);
+    Query query = new Query("Comment").addSort("rate", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    ArrayList<String> likedOptionsVal = new ArrayList<>();
+    // Add all the comments in the datastore to the comments list
+    for (Entity entity : results.asIterable()) {
+        String authorVal = (String) entity.getProperty("author");
+        int rateVal = (int)(long) entity.getProperty("rate");
+        likedOptionsVal.clear();
+        if(((String) entity.getProperty("is_info_liked")).equals("true")){
```

Comment:
Suggestion: replace with:

if (Boolean.parseBoolean((String) entity.getProperty("is_info_liked")))
Same for all other boolean properties